src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+        }
+        return thisStake.liquidity;
+    }
+
+    function getReward(address account)
+        public
+        nonReentrant
+        updateReward(account, true)
+        onlyJarAndAuthorised
+        returns (uint256 _rewardsForJar)
+    {
+        uint256 reward;
+
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            reward = _rewards[account][i];
+            if (reward > 0) {
+                _rewards[account][i] = 0;
+                _rewardsForJar += reward;
+                IERC20(rewardTokens[i]).safeTransfer(account, reward);
+                emit RewardPaid(account, rewardSymbols[i], reward);
+            }
+        }
+    }
+
+    function exit(address account) external returns(uint256 withdrawAmount){
+        withdrawAmount = withdrawAll(account);
+        getReward(account);
+    }
+
+    /* ========== RESTRICTED FUNCTIONS ========== */
+
+    function notifyRewardAmount(uint256[] memory rewards)
+        external
+        onlyDistribution
+        updateReward(address(0), false)
+    {
+        require(
+            rewards.length == rewardTokens.length,
+            "Rewards count do not match reward token count"
+        );
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            IERC20(rewardTokens[i]).safeTransferFrom(
+                DISTRIBUTION,
+                address(this),
+                rewards[i]
+            );
+
+            if (block.timestamp >= periodFinish) {
+                rewardRates[i] = rewards[i] / DURATION;
+            } else {
+                uint256 remaining = periodFinish - block.timestamp;
+                uint256 leftover = remaining * rewardRates[i];
+                rewardRates[i] = (rewards[i] + leftover) / DURATION;
+            }
+
+            // Ensure the provided reward amount is not more than the balance in the contract.
+            // This keeps the reward rate in the right range, preventing overflows due to
+            // very high values of rewardRate in the earned and rewardsPerToken functions;
+            // Reward + leftover must be less than 2^256 / 10^18 to avoid overflow.
+            uint256 balance = IERC20(rewardTokens[i]).balanceOf(address(this));
+            require(
+                rewardRates[i] <= balance / DURATION,
+                "Provided reward too high"
+            );
+
+            emit RewardAdded(rewardSymbols[i], rewards[i]);
+        }
+
+        lastUpdateTime = block.timestamp;
+        periodFinish = block.timestamp + DURATION;
+    }
+
+    function setMultipliers(uint256 _lock_max_multiplier) external onlyGov {
+        require(
+            _lock_max_multiplier >= uint256(1e18),
+            "Multiplier must be greater than or equal to 1e18"
+        );
+        lockMaxMultiplier = _lock_max_multiplier;
+        emit LockedStakeMaxMultiplierUpdated(lockMaxMultiplier);
+    }
+
+    function setMaxRewardsDuration(uint256 _lockTimeForMaxMultiplier)
```

Comment:
I think these functions `setMaxRewardsDuration` and `setMultipliers` are dangerous because it disturbs existing `derivedSupply` and `derivedBalance` calculations by naively changing critical global variables. I honestly think these are better off removed.